lib/token-request.js

```diff
@@ -163,6 +163,22 @@ TokenRequest.prototype._getTokenWithCacheWrapper = function(callback, getTokenFu
   });
 };
 
+TokenRequest.prototype._storeTokenWithCacheWrapper = function(callback, getTokenFunc) {
```

Comment:
I think this function is more complicated than it needs to be.  I like factoring out the code that adds the token to the cache, however, in this case you do not need to include the getToken call in the same function that adds the token to the cache.  Leave that bit of code in getTokenWithDeviceCode.  When the getToken function returns, or calls the callback, then call a separate function to add the token to the cache.  Something like:

``` js
 this._oauthGetTokenByPolling(oauthParameters, interval, expires_in, function(err) {
   if (err) {
  } else {
   addTokenToCache(callback)
  } );
```